### PR TITLE
Enabled masked fill and scatter for bool tensors

### PR DESCRIPTION
**Pull request by izdeby**
Enabled masked_fill, masked_fill_, masked_scatter_, masked_scatter on bool tensors. 
Tested via unit tests

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -8053,6 +8066,17 @@ def test_masked_fill(self):
...
             self.assertEqual(dst, dst2, 0)
 
+    def test_masked_fill_bool_tensor(self):
+        for device in torch.testing.get_all_device_types():
+            dst = torch.tensor([True, False, True], device=device)
+            mask = torch.tensor([False, True, False], device=device)
+
+            dst.masked_fill_(mask, True)
+            self.assertEqual(dst, torch.tensor([True, True, True], device=device))
+
```
Comment (VitalyFedyunin) [inline test/test_torch.py]: Super nit: you might recreate dst here, for readability.

**Review by VitalyFedyunin — APPROVED** [review]
